simple_history/utils.py

```diff
@@ -1,11 +1,184 @@
+from django.db.models import Case, ForeignKey, ManyToManyField, Model, Q, When
 from django.forms.models import model_to_dict
 
-from simple_history.exceptions import AlternativeManagerError, NotHistoricalModelError
+from .exceptions import AlternativeManagerError, NotHistoricalModelError
+
+if TYPE_CHECKING:
+    from .manager import HistoricalQuerySet, HistoryManager
+    from .models import HistoricalChanges
+
+
+@contextmanager
+def disable_history(
+    *,
+    only_for_model: Type[Model] = None,
+    instance_predicate: Callable[[Model], bool] = None,
+) -> Iterator[None]:
+    """
+    Disable creating historical records while this context manager is active.
+
+    Note: ``only_for_model`` and ``instance_predicate`` cannot both be provided.
+
+    :param only_for_model: Only disable history creation for this model type.
+    :param instance_predicate: Only disable history creation for model instances passing
+        this predicate.
+    """
+    assert (  # nosec
+        _StoredDisableHistoryInfo.get() is None
+    ), "Nesting 'disable_history()' contexts is undefined behavior"
+
+    if only_for_model:
+        # Raise an error if it's not a history-tracked model
+        get_history_manager_for_model(only_for_model)
+        if instance_predicate:
+            raise ValueError(
+                "'only_for_model' and 'instance_predicate' cannot both be provided"
+            )
+        else:
+
+            def instance_predicate(instance: Model):
+                return isinstance(instance, only_for_model)
+
+    info = _StoredDisableHistoryInfo(instance_predicate)
+    info.set()
+    try:
+        yield
+    finally:
+        info.delete()
+
+
+@dataclass(frozen=True)
+class _StoredDisableHistoryInfo:
+    """
+    Data related to how historical record creation is disabled, stored in
+    ``HistoricalRecords.context`` through the ``disable_history()`` context manager.
+    """
+
+    LOCAL_STORAGE_ATTR_NAME: ClassVar = "disable_history_info"
+
+    instance_predicate: Callable[[Model], bool] = None
+
+    def set(self) -> None:
+        setattr(self._get_storage(), self.LOCAL_STORAGE_ATTR_NAME, self)
+
+    @classmethod
+    def get(cls) -> Optional["_StoredDisableHistoryInfo"]:
+        """
+        A return value of ``None`` means that the ``disable_history()`` context manager
+        is not active.
+        """
+        return getattr(cls._get_storage(), cls.LOCAL_STORAGE_ATTR_NAME, None)
+
+    @classmethod
+    def delete(cls) -> None:
+        delattr(cls._get_storage(), cls.LOCAL_STORAGE_ATTR_NAME)
+
+    @staticmethod
+    def _get_storage() -> Local:
+        from .models import HistoricalRecords  # Avoids circular importing
+
+        return HistoricalRecords.context
+
+
+@dataclass(
+    frozen=True,
+    # DEV: Replace this with just `kw_only=True` when the minimum required
+    #      Python version is 3.10
+    **({"kw_only": True} if sys.version_info >= (3, 10) else {}),
+)
+class DisableHistoryInfo:
```

Comment:
I didn't have an issue with this before, but looking at the first method of `not_disabled` and I'm not sure this is how things should be done. My concern is that this control is solely focused on history being disabled. However, that also means it's focused on history being enabled.

So maybe a name like `HistoryControl`, `HistoryState` or `HistoryInstrumentation`? Then we can change `not_disabled` to `enabled` :grin: